service/frontend/workflowHandler.go

```diff
@@ -1686,355 +1686,6 @@ func (wh *WorkflowHandler) StartWorkflowExecution(
-			tag.WorkflowID(getRequest.Execution.GetWorkflowId()),
-			tag.WorkflowRunID(getRequest.Execution.GetRunId()),
-			tag.WorkflowDomainID(domainID), tag.WorkflowSize(int64(getRequest.GetMaximumPageSize())))
-		getRequest.MaximumPageSize = common.Int32Ptr(common.GetHistoryMaxPageSize)
-	}
-
-	// this function return the following 5 things,
-	// 1. current branch token
-	// 2. the workflow run ID
-	// 3. the next event ID
-	// 4. error if any
-	queryHistory := func(
-		domainUUID string,
-		execution *gen.WorkflowExecution,
-		currentBranchToken []byte,
-	) ([]byte, string, int64, error) {
-		response, err := wh.GetHistoryClient().GetMutableState(ctx, &h.GetMutableStateRequest{
-			DomainUUID:          common.StringPtr(domainUUID),
-			Execution:           execution,
-			ExpectedNextEventId: nil,
-			CurrentBranchToken:  currentBranchToken,
-		})
-
-		if err != nil {
-			return nil, "", 0, err
-		}
-
-		return response.CurrentBranchToken,
-			response.Execution.GetRunId(),
-			response.GetNextEventId(),
-			nil
-	}
-
-	execution := getRequest.Execution
-	token := &getHistoryContinuationToken{}
-
-	var runID string
-	var nextEventID int64
-
-	// process the token for paging
-	if getRequest.NextPageToken != nil {
-		token, err = deserializeHistoryToken(getRequest.NextPageToken)
-		if err != nil {
-			return nil, wh.error(errInvalidNextPageToken, scope)
-		}
-		if execution.RunId != nil && execution.GetRunId() != token.RunID {
-			return nil, wh.error(errNextPageTokenRunIDMismatch, scope)
-		}
-
-		execution.RunId = common.StringPtr(token.RunID)
-
-	} else {
-
-		token.BranchToken, runID, nextEventID, err =
-			queryHistory(domainID, execution, nil)
-		if err != nil {
-			return nil, wh.error(err, scope)
-		}
-
-		execution.RunId = &runID
-
-		token.RunID = runID
-		token.FirstEventID = common.FirstEventID
-		token.NextEventID = nextEventID
-		token.PersistenceToken = nil
-	}
-
-	history := []*gen.DataBlob{}
-	// return all events
-	if token.FirstEventID >= token.NextEventID {
-		return &gen.GetWorkflowExecutionRawHistoryResponse{
-			RawHistory:    []*gen.DataBlob{},
-			NextPageToken: nil,
-		}, nil
-	}
-
-	history, token.PersistenceToken, err = wh.getRawHistory(
-		scope,
-		domainID,
-		*execution,
-		token.FirstEventID,
-		token.NextEventID,
-		getRequest.GetMaximumPageSize(),
-		token.PersistenceToken,
-		token.TransientDecision,
-		token.BranchToken,
-	)
-	if err != nil {
-		return nil, wh.error(err, scope)
-	}
-
-	if len(token.PersistenceToken) == 0 {
-		// meaning, there is no more history to be returned
-		token = nil
-	}
-
-	nextToken, err := serializeHistoryToken(token)
-	if err != nil {
-		return nil, wh.error(err, scope)
-	}
-	return &gen.GetWorkflowExecutionRawHistoryResponse{
-		RawHistory:    history,
-		NextPageToken: nextToken,
-	}, nil
-}
-
-// PollForWorkflowExecutionRawHistory get Raw History for long poll request
-func (wh *WorkflowHandler) PollForWorkflowExecutionRawHistory(
```

Comment:
why was there a PollForWorkflowExecutionRawHistory but no corresponding PollForWorkflowExecutionHistory ? Why did we create this API ?

Reply:
> why was there a PollForWorkflowExecutionRawHistory but no corresponding PollForWorkflowExecutionHistory ? Why did we create this API ?

PollForWorkflowExecutionRawHistory was added by Anish for long pull of raw history purposed. Since we put all logic in one common API, we do not need it any more. We do not have PollForWorkflowExecutionHistory, the long pull implementation also resides in the GetWorkflowExecutionHistory API.

Reply:
After an initial conversation with @yycptt @yux0, we wanted to separate GetWorkflowExecution API into two API, long pull and short pull, to make the code simpler and also to have different metrics for long pull and short pull.